packages/nvidia_nat_app/tests/graph/test_static_analysis.py

```
@@ -0,0 +1,472 @@
+    def test_dict_read_via_get(self):
+
+        def fn(state):
+            x = state.get("query")
+            return {"result": x}
+
+        r = analyze_function_ast(fn)
+        assert "query" in r.reads.all_fields_flat
+
+    def test_dict_write_via_return(self):
+
+        def fn(state):
```

Ruff (0.15.2)

[warning] Unused function argument: `state`

(ARG001)